autoextract_poet/page_inputs.py

```diff
@@ -0,0 +1,54 @@
+from typing import ClassVar, Optional, Type
+
+import attr
+
+from autoextract_poet.items import (
+    Article,
+    Item,
+    Product,
+)
+
+
+@attr.s(auto_attribs=True)
+class _AutoExtractData:
+    """Container for AutoExtract data.
+
+    Should not be used directly by providers.
+    Use derived classes like AutoExtractArticleData and similar.
+
+    API responses are wrapped in a JSON array
+    (this is to facilitate query batching)
+    but we're receiving single responses here..
+
+    https://doc.scrapinghub.com/autoextract.html#responses
+    """
+
+    item_class: ClassVar[Type[Item]]
+    item_key: ClassVar[str]
```

Comment:
I'm still unsure about this `item_key`. The question is if `data` should be `{"article": {...}}` dict or just a dict with article data (`{...}`, i.e. provider calls `data["article"]` itself before creating AutoExtractArticleData instance).

As we discussed on a call, escape hatch (e.g. to get html) may be unnecessary, as it can be solved in providers.

So this leaves us with an use case where AutoExtractArticleData (or a similar class) needs to access several top-level fields to create an item. I think that's a valid use case. For example, why not provide page language in addition to article language right here. But in this case `item_key` is not needed, because to_item is going to access several keys. 

So, a proposal: keep the current approach, but make `item_key` private, rename it to `_item_key`. It looks like an implementation detail, not a part of API intended to be exposed.

---

An additional argument for not requiring `{"article": {...}}` in `data` dicts: if article is missing from the response for some reason, it would fail earlier: in a provider, not in to_item method. Provider would need to do this check explicitly with the current approach, to fail earlier. However, I guess we can discuss & fix this later; it'd be backwards incompatible, but that can be ok. For now my proposal is to just make item_key private.

Reply:
I've renamed the `item_key` class attribute into `_item_key` for now.

Reply:
On the same way, shouldn't we make `item_class` private as well?

Reply:
> On the same way, shouldn't we make item_class private as well?

I'm fine with both. Though it could make sense to make it private now, as you're suggesting, because changing it back to public is easier than the other way around.

Reply:
@kmike, I've reverted the `item_key` to a public class attribute since it's going to be useful for other libraries when implementing providers for those Page Inputs.

Reply:
And `item_class` was converted to a property that uses the information passed to the generic type T.